src/client/InteractionClient.js

```diff
@@ -0,0 +1,232 @@
+    const commands = await path.commands.get();
+    return commands.map(c => new ApplicationCommand(this, c, guildID));
+  }
+
+  /**
+   * Set all the commands for the application or guild.
+   * @param {Object[]} commands The command descriptor.
+   * @param {Snowflake} [guildID] Optional guild ID.
+   * @returns {ApplicationCommand[]} The commands.
+   */
+  async setCommands(commands, guildID) {
+    let path = this.client.api.applications(this.applicationID);
+    if (guildID) {
+      path = path.guilds(guildID);
+    }
+    const cs = await path.commands.post({
+      data: commands.map(transformCommand),
+    });
+    return cs.map(c => new ApplicationCommand(this, c, guildID));
+  }
+
+  /**
+   * Create a command.
+   * @param {Object} command The command descriptor.
+   * @param {Snowflake} [guildID] Optional guild ID.
+   * @returns {ApplicationCommand} The created command.
+   */
+  async createCommand(command, guildID) {
+    let path = this.client.api.applications(this.clientID);
+    if (guildID) {
+      path = path.guilds(guildID);
+    }
+    const c = await path.commands.post({
+      data: transformCommand(command),
+    });
+    return new ApplicationCommand(this, c, guildID);
+  }
+
+  handle(data) {
+    switch (data.type) {
+      case InteractionType.PING:
+        return {
+          type: InteractionResponseType.PONG,
+        };
+      case InteractionType.APPLICATION_COMMAND: {
+        let timedOut = false;
+        let resolve;
+        const directPromise = new Promise(r => {
+          resolve = r;
+          this.client.setTimeout(() => {
+            timedOut = true;
+            r({
+              type: InteractionResponseType.DEFERRED_CHANNEL_MESSAGE_WITH_SOURCE,
+            });
+          }, 250);
+        });
+
+        const syncHandle = {
+          acknowledge() {
+            if (!timedOut) {
+              resolve({
+                type: InteractionResponseType.DEFERRED_CHANNEL_MESSAGE_WITH_SOURCE,
+              });
+            }
+          },
+          reply(resolved) {
+            if (timedOut) {
+              return false;
+            }
+            resolve({
+              type: InteractionResponseType.CHANNEL_MESSAGE_WITH_SOURCE,
+              data: resolved.data,
+            });
+            return true;
+          },
+        };
+
+        const interaction = new CommandInteraction(this.client, data, syncHandle);
+
+        /**
+         * Emitted when an interaction is created.
+         * @event Client#interactionCreate
+         * @param {Interaction} interaction The interaction which was created.
+         */
+        this.client.emit(Events.INTERACTION_CREATE, interaction);
+
+        return directPromise;
+      }
+      default:
+        throw new RangeError('Invalid interaction data');
+    }
+  }
+
+  /**
+   * An express-like middleware factory which can be used
+   * with webhook interactions.
+   * @returns {Function} The middleware function.
+   */
+  middleware() {
```

Comment:
This is BASICALLY an express middleware, so we should document it as such... Yes, other modules may have the same syntax, but at its core that's what it's meant for

```suggestion
   * An express middleware factory which can be used
   * with webhook interactions.
   * @returns {Function} The middleware function.
   */
  expressMiddleware() {
```